public/docs/ts/latest/guide/architecture.jade

```diff
@@ -193,196 +221,214 @@ figure
-    to separate the component from the directive in our architectural overview.
+    While **a component is technically a directive**,
+    components are so distinctive and central to Angular applications that we chose
+    to separate components from directives in this architectural overview.
 :marked
-  There are two *other* kinds of directives as well that we call "structural" and "attribute" directives.
-  
-  They tend to appear within an element tag like attributes, 
+  Two *other* kinds of directives exist: _structural_ and _attribute_ directives.
+
+  They tend to appear within an element tag like attributes,
   sometimes by name but more often as the target of an assignment or a binding.
-  
+
   **Structural** directives alter layout by adding, removing, and replacing elements in DOM.
-  
-  We see two built-in structural directives at play in our [example](#template) template:
-+makeExample('architecture/ts/app/hero-list.component.1.html', 'structural')(format=".")
+
+  Our [example](#templates) template uses two built-in structural directives:
+
++makeExcerpt('app/hero-list.component.1.html', 'structural')
+
 :marked
-  * [`*ngFor`](displaying-data.html#ngFor) tells Angular to stamp out one `<div>` per hero in the `heroes` list.
+  * [`*ngFor`](displaying-data.html#ngFor) tells Angular to stamp out one `<li>` per hero in the `heroes` list.
   * [`*ngIf`](displaying-data.html#ngIf) includes the `HeroDetail` component only if a selected hero exists.
-  
-  **Attribute** directives alter the appearance or behavior of an existing element. 
-  In templates they look like regular HTML attributes, hence the name.
-  
-  The `ngModel` directive, which implements two-way data binding, is an example of an attribute directive.
-+makeExample('architecture/ts/app/hero-detail.component.html', 'ngModel')(format=".")
+
+.l-sub-section
+  :marked
+    In Dart, **the only value that is true is the boolean value `true`**; all
+    other values are false. JavaScript and TypeScript, in contrast, treat values
+    such as 1 and most non-null objects as true. For this reason, the JavaScript
+    and TypeScript versions of this app can use just `selectedHero` as the value
+    of the `*ngIf` expression. The Dart version must use a boolean operator such
+    as `!=` instead.
+
 :marked
-  It modifies the behavior of an existing element (typically an `<input>`) 
+  **Attribute** directives alter the appearance or behavior of an existing element.
+  In templates they look like regular HTML attributes, hence the name.
+
+  The `ngModel` directive, which implements two-way data binding, is
+  an example of an attribute directive. `ngModel` modifies the behavior of
+  an existing element (typically an `<input>`)
   by setting its display value property and responding to change events.
-  
-  Angular ships with a few other directives that either alter the layout structure 
+
++makeExcerpt('app/hero-detail.component.html', 'ngModel')
```

Comment:
I just realized that a lot of code snippets have headings that didn't used to. Do we really need the file info? I'd rather just concentrate on the code, for small snippets like this where the filename seems irrelevant.